Add docs for proxies (#2544)
rust-lang/rustup: doc/src/proxies.md

# Proxies

> `rustup` provides proxy wrapper through [components](https://github.com/rust-lang/rustup/blob/master/doc/src/concepts/components.md) and you can see the list of the proxies [here](https://github.com/rust-lang/rustup/blob/22dc71adbd7e19de0090f991b3609e5f14910dcb/src/lib.rs#L17-L28)

Reviewer: This might read slightly better as something like:

```markdown
`rustup` provides a number of wrappers for common Rust tools.
These are called _proxies_ and represent commands which are
provided by the various [components][].

The list of proxies is currently static in `rustup` and is as follows:

[components]: concepts/components.html
```

Author: @kinnison I appreciate your feedback. I agree that moving the doc under the `concepts/` subtree is more descriptive and clear.

> - `rustc`: `rustc` is a proxy for the compiler for the Rust programming language, provided by the project itself and it matches `rustc` component.

Reviewer: I think, given this is explicitly already a list of proxies, we can simplify the language here a little:

```suggestion
- `rustc` is the compiler for the Rust programming language, provided by the project itself and comes from the `rustc` component.
```

Note how I don't say "proxy" and I don't bother repeating the name of the proxy either, there's simply no need.

Reviewer: You can make similar changes to all the lines.  No need to keep saying "proxy"

> - `rustdoc`: `rustdoc` is a proxy for the standard Rust distribution ships with a tool called rustdoc that generates documentation for Rust projects and it matches `rustc` component.

Reviewer: Again the language here is very complex, a simpler example could be:

```suggestion
- `rustdoc` is a tool distributed in the `rustc` component which helps you to generate documentation for Rust projects.
```

Reviewer: Similar simplifications can be made to other entries.

In general, try not to be over-complex in your language or description, simpler is better at this point.
